cmd/treedb_vector_partition_bench/hc_bridge.go

```
@@ -0,0 +1,288 @@
+		if _, err := fmt.Sscanf(neighbor.ID, "doc-%06d", &n); err != nil || fmt.Sprintf("doc-%06d", n) != neighbor.ID {
+			hcBridgeWriteErrorV1(w, http.StatusBadGateway, "invalid_neighbor_id")
+			return
+		}
+		ids[i] = n
+	}
+	if len(response.Search.Neighbors) != request.TopK || response.Search.Counters.SelectedPartitions != uint64(request.Probes) || response.Search.Counters.SelectedPartitions == 0 || response.Search.Counters.HNSWServedPartitions > response.Search.Counters.SelectedPartitions || response.Search.Counters.ExactScanPartitions != response.Search.Counters.SelectedPartitions-response.Search.Counters.HNSWServedPartitions {
+		hcBridgeWriteErrorV1(w, http.StatusBadGateway, "incomplete_route_proof")
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(struct {
```

golangci-lint (2.12.2)

[error] Error return value of `(*encoding/json.Encoder).Encode` is not checked: unsafe type `float32` found

(errchkjson)